kyaml/runfn/runfn.go

```diff
@@ -91,7 +91,6 @@ func (r *RunFns) init() {
 	if r.containerFilterProvider == nil {
 		r.containerFilterProvider = func(image, path string, api *yaml.RNode) kio.Filter {
 			cf := &filters.ContainerFilter{Image: image, Config: api}
-			cf.SetMountPath(filepath.Join(r.Path, path))
```

Comment:
I created this [method](https://github.com/kubernetes-sigs/kustomize/pull/1822/commits/e46108ada0c22c011d8d66f64969a84b6fccbbe4#diff-2ccfa1b7748a274f1d1f1de52b35038eR65) to replace SetMountPath()'s usage but wanted second opinions on whether or not it should be used in this specific context.

Reply:
This is fine, I think we want to mount the directory containing the configuration function by default, but this should probably go in the cobra command rather than the library.

We *should* add StorageMounts to RunFns, and then copy it to the ContainerFilter to expose that part of the API:

```go
type RunFns struct {
...
  StorageMounts []StorageMount
...
}
```

Reply:
> We should add StorageMounts to RunFns, and then copy it to the ContainerFilter to expose that part of the API

I couldn't tell if this [revision](https://github.com/kubernetes-sigs/kustomize/pull/1822/commits/dff30b926e18acbe8c6be1e620896d90f78293d3#diff-e07a50718371004dc2235f9638ea10d0R95) is what was intended by adding StorageMounts to RunFns and then copying it to the ContainerFilter. Code also below: 

```go
defaultMount := filters.StorageMount{}
r.StorageMounts = append(r.StorageMounts, defaultMount)
cf := &filters.ContainerFilter{Image: image, Config: api, StorageMounts: r.StorageMounts}
```